### PR TITLE
textfields [1 of 3]: add text into speech bubble; also add rich text example

### PR DESCRIPTION
**Pull request by mimecuvalo**
This is the first of three textfield changes. This starts with making the speech bubble actually have text. Also, it creates a TipTap example and how that would be wired up.

🎵 this is dangerous, I walk through textfields so watch your head rock 🎵 

### Change Type

- [x] `minor` — New feature

### Release Notes

- Refactor textfields be composable/swappable.

### CODE REVIEW

**Comment by SomeHats** [comment]
If we're going to add these examples, I'd like them to demonstrate a slightly more complete idea of what adding text to a shape might involve and the type of bugs you might run into e.g.

How should the shape change as the text grows?
Kapture 2024-03-04 at 12 24 42

Focus management is hard
Kapture 2024-03-04 at 12 29 57

File: packages/tldraw/src/lib/shapes/text/TextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	rInput,
+	text,
+	handleFocus,
+	handleChange,
+	handleKeyDown,
+	handleBlur,
+	handleInputPointerDown,
+	handleDoubleClick,
+}: {
+	rInput: React.RefObject<HTMLTextAreaElement>
```
Comment (steveruizok) [inline packages/tldraw/src/lib/shapes/text/TextArea.tsx:13]: forward ref instead here?

Comment (steveruizok) [reply packages/tldraw/src/lib/shapes/text/TextArea.tsx:13]: (I mean they're taking it out in a few months anyway, but for now!)

**Comment by steveruizok** [comment]
Not sure what the priorities are set to, but could we push the more complex speech bubble example down?

image

**Comment by mimecuvalo** [comment]
- fixed up growing for shapes. the rich text one isn't 100% perfect for every edge case but it's meant to be an example so I figure it doesn't have to reach 100% there, it's meant to be a demo.
- fixed up focus management, good flag @SomeHats 
- fixed up the forwardRef, doh - thx @steveruizok 
- changed priority of examples

File: packages/tldraw/src/lib/shapes/text/TextArea.tsx (hunk 1 above)
Comment (mimecuvalo) [reply packages/tldraw/src/lib/shapes/text/TextArea.tsx:13]: hah doh, not sure what i was smoking here - reworked!

**Comment by mimecuvalo** [comment]
The refactor work in this PR would help support https://github.com/tldraw/tldraw/pull/3132 (another reason I was working on textfields)
We should land this first PR at least, please! @steveruizok it's waiting your approval ✅ - thx!

**Comment by steveruizok** [comment]
Let's split this up!

Let's land:
- the changes that turn our <textareas> into a TextArea component
- the simple example

Let's not land:
- the changes to TextEditor
- the rich text example

**Review by steveruizok — APPROVED** [review]

**Pull request merged**